client/src/components/aggregations/AggregationWidgets.js

```diff
@@ -0,0 +1,296 @@
+        minute: "2-digit",
+        meridiem: false,
+        omitZeroMinute: false,
+        hour12: false
+      }}
+      height="auto" // assume a natural height, no scrollbars will be used
+      aspectRatio={3} // ratio of width-to-height
+      ref={calendarComponentRef}
+      events={events}
+      eventOverlap
+      eventLimit
+    />
+  )
+}
+CalendarWidget.propTypes = {
+  hasPrevNext: PropTypes.bool,
+  ...aggregationWidgetPropTypes
+}
+CalendarWidget.defaultProps = {
+  hasPrevNext: false,
+  ...aggregationWidgetDefaultProps
+}
+
+export const DefaultAggWidget = ({
+  values,
+  whenUnspecified,
+  ...otherWidgetProps
+}) => {
+  const [showValues, setShowValues] = useState(false)
+  const filteredValues = values.filter(value => !utils.isNullOrUndefined(value))
+  if (_isEmpty(filteredValues)) {
+    return whenUnspecified
+  }
+  return (
+    <div>
+      <Button
+        className="toggle-section-button"
+        style={{ marginBottom: "1rem" }}
+        onClick={toggleShowValues}
+        id="toggleShowValues"
+      >
+        {showValues ? "Hide" : "Show"} {filteredValues.length} values
+      </Button>
+      <Collapse in={showValues}>
+        <Table>
+          <tbody>
+            {filteredValues.map(val => {
+              const keyValue = _uniqueId("value_")
+              return (
+                <tr key={keyValue}>
+                  <td>{val}</td>
+                </tr>
+              )
+            })}
+          </tbody>
+        </Table>
+      </Collapse>
+    </div>
+  )
+  function toggleShowValues() {
+    setShowValues(!showValues)
+  }
+}
+DefaultAggWidget.propTypes = aggregationWidgetPropTypes
+DefaultAggWidget.defaultProps = aggregationWidgetDefaultProps
+
+export const ReportsMapWidget = ({
+  values,
+  mapId,
+  width,
+  height,
+  whenUnspecified,
+  ...otherWidgetProps
+}) => {
+  const markers = useMemo(() => {
+    if (!values.length) {
+      return []
+    }
+    const markerArray = []
+    values.forEach(report => {
+      if (Location.hasCoordinates(report.location)) {
+        let label = _escape(report.intent || "<undefined>") // escape HTML in intent!
+        label += `<br/>@ <b>${_escape(report.location.name)}</b>` // escape HTML in locationName!
+        markerArray.push({
+          id: report.uuid,
+          lat: report.location.lat,
+          lng: report.location.lng,
+          name: label
+        })
+      }
+    })
+    return markerArray
+  }, [values])
+  if (_isEmpty(markers)) {
+    return whenUnspecified
+  }
+  return (
+    <div className="non-scrollable">
+      <Leaflet
+        markers={markers}
+        width={width}
+        height={height}
+        mapId={mapId}
+        marginBottom={0}
+      />
+    </div>
+  )
+}
+ReportsMapWidget.propTypes = {
+  ...aggregationWidgetPropTypes,
+  mapId: PropTypes.string,
+  width: PropTypes.number,
+  height: PropTypes.number
+}
+ReportsMapWidget.defaultProps = {
+  values: [],
+  ...aggregationWidgetDefaultProps
+}
```

Comment:
Should this file perhaps be split up into individual widgets?

Reply:
Done